pi4j-test/src/test/java/com/pi4j/test/smoketest/provider/ProviderSmokeTest.java

```
@@ -0,0 +1,264 @@
+ *  #L%
+ *
+ *
+ */
+
+import com.pi4j.Pi4J;
+import com.pi4j.boardinfo.util.PwmChipUtil;
+import com.pi4j.context.Context;
+import com.pi4j.exception.Pi4JException;
+import com.pi4j.io.gpio.digital.*;
+import com.pi4j.io.i2c.I2C;
+import com.pi4j.io.pwm.Pwm;
+import com.pi4j.io.pwm.PwmType;
+import com.pi4j.io.spi.Spi;
+import com.pi4j.io.spi.SpiBus;
+import com.pi4j.io.spi.SpiChipSelect;
+import com.pi4j.io.spi.SpiMode;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+ /**
+  * ProviderSmokeTest
+  * Simple test of the six providers. Dependent upon
+  * the wiring described in the README file.
+  *
+  */
+ public class ProviderSmokeTest {
+
+    static Context pi4j = null;
+
+    static int pwmFlashes = 0 ;
+
+    @BeforeAll
+    public static void beforeTest() {
+
+        System.setProperty(org.slf4j.simple.SimpleLogger.DEFAULT_LOG_LEVEL_KEY, "TRACE");
+        pi4j = Pi4J.newAutoContext();
+
+    }
+
+
+    @AfterAll
+    public static void afterTest() {
+        try {
+            pi4j.shutdown();
+        } catch (Pi4JException e) { /* do nothing */ }
+    }
+
+
+    @Test
+    public void testI2c() {
+        final int idValueMskBMP = 0x58;   // expected chpId value BMP280
+        final int idValueMskBME = 0x60;   // expected chpId value BME280
+
+        I2C dev = createI2cBMPDevice();
+        // read 0xD0 validate data equal 0x58 or 0x60
+        int id = dev.readRegister(0xD0);
+
+        Assertions.assertTrue(id == idValueMskBMP || id ==idValueMskBME, "TestBMP and BME ID");
+    }
+
+    @Test
+    public void testSpi() {
+        final int idValueMskBMP = 0x58;   // expected chpId value BMP280
+        final int idValueMskBME = 0x60;   // expected chpId value BME280
+
+        Spi spi = createSPIDevice();
+        final int chipId = 0xD0;
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException e) {
```

Just let the test throw, Junit will take care...